### DISCUSSION TITLE
It's possible to not run every version in the matrix?

### DISCUSSION THREAD

**Discussion by luchiago** [community/community#25406] [Actions]
<p>I want to not run every version of python settled at the matrix depending on the file that was modified on the pull request. This could be done?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/luchiago">@luchiago</a> , </p>
<p>There is a new function <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#fromjson" rel="nofollow noopener">fromJson()</a> to dynamicly set matrix setting.</p>
<p>You could add another job to get the changed file on the pull request and then set matrix setting depending on it. </p>
<p>There is a previous ticket for finding pull request file changes : <a href="https://github.community/t5/GitHub-Actions/Get-list-of-files-on-pull-request-merge/td-p/41570" rel="nofollow noopener">https://github.community/t5/GitHub-Actions/Get-list-of-files-on-pull-request-merge/td-p/41570 </a></p>
<p>Then you could follow my example workflow to set python version in matrix .</p>
<pre><code>jobs:
  getchange:
   outputs:
      matrix: ${{ steps.changes.outputs.matrix }}
   runs-on: ubuntu-latest
   steps:
     - id: changes
       ## set matrix value based on file changes
       run: |
         ....... // write your script to get file change then set matrix value
         echo "::set-output name=matrix::{\"include\":[{\"python\":\"3.5\" },{\"python\":\"3.7\"}]}"
  build:
    needs: [getchange]
    runs-on: ubuntu-latest
    strategy:
      matrix: ${{ fromJson(needs.getchange.outputs.matrix) }}

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python }}
      uses: actions/setup-python@v1
      with:
        python-version: ${{ matrix.python }}
</code></pre>

**Comment by luchiago** [comment]
<p>Thanks, <a class="mention" href="/yanjingzhu">@yanjingzhu</a>, I will implement your suggestion and see if fits the project and give you feedback about it.</p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/luchiago">@luchiago</a>,<br>
Does fromJson() function work for you?  Is there any other questions you need help?</p>

**Comment by luchiago** [comment]
<p>Yes, it did. Thank you so much for your help. I don’t have any other questions.</p>
